test/performance/odp_packet_gen.c

```diff
@@ -1449,13 +1620,15 @@ static int init_packets(test_global_t *global, int pktio,
 		eth = data;
 		memcpy(eth->dst.addr, global->pktio[pktio].eth_dst.addr, 6);
 		memcpy(eth->src.addr, global->pktio[pktio].eth_src.addr, 6);
-		eth->type = odp_cpu_to_be_16(ODPH_ETHTYPE_IPV4);
+		eth->type = odp_cpu_to_be_16(test_options->eth_type);
 		l2_len = ODPH_ETHHDR_LEN;
+		odp_packet_has_eth_set(pkt, 1);
 
 		/* VLAN(s) */
 		if (num_vlan) {
 			tpid = test_options->vlan[0].tpid;
 			eth->type = odp_cpu_to_be_16(tpid);
+			odp_packet_has_vlan_set(pkt, 1);
```

Comment:
Depending on TPID value could also be VLAN QinQ.

Reply:
V4.